internal/auth0/user/resource_permission.go

```diff
@@ -0,0 +1,169 @@
+			},
+		},
+		CreateContext: createUserPermission,
+		ReadContext:   readUserPermission,
+		DeleteContext: deleteUserPermission,
+		Importer: &schema.ResourceImporter{
+			StateContext: importUserPermission,
+		},
+		Description: "With this resource, you can manage user permissions.",
+	}
+}
+
+func createUserPermission(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+
+	userId := data.Get("user_id").(string)
+	resourceServerId := data.Get("resource_server_identifier").(string)
+	permissionName := data.Get("permission").(string)
+
+	mutex.Global.Lock(userId)
+	defer mutex.Global.Unlock(userId)
+
+	if err := api.User.AssignPermissions(userId, []*management.Permission{
+		{
+			ResourceServerIdentifier: &resourceServerId,
+			Name:                     &permissionName,
+		},
+	}); err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(userId + resourceServerId + permissionName)
+
+	return readUserPermission(ctx, data, meta)
+}
+
+func readUserPermission(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+
+	userId := data.Get("user_id").(string)
+	permissionName := data.Get("permission").(string)
+	resourceServerId := data.Get("resource_server_identifier").(string)
+
+	existingPermissions, err := api.User.Permissions(userId)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	for _, p := range existingPermissions.Permissions {
+		if p.GetName() == permissionName && p.GetResourceServerIdentifier() == resourceServerId {
+			data.Set("description", p.GetDescription())
+			data.Set("resource_server_name", p.GetResourceServerName())
+			return nil
+		}
+	}
+
+	data.SetId("")
+	return nil
+
+}
+
+func deleteUserPermission(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+
+	userId := data.Get("user_id").(string)
+	permissionName := data.Get("permission").(string)
+	resourceServerId := data.Get("resource_server_identifier").(string)
+
+	mutex.Global.Lock(userId)
+	defer mutex.Global.Unlock(userId)
+
+	if err := api.User.RemovePermissions(
+		userId,
+		[]*management.Permission{
+			{
+				ResourceServerIdentifier: &resourceServerId,
+				Name:                     &permissionName,
+			},
+		},
+	); err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	data.SetId("")
+	return nil
+}
+
+func importUserPermission(
+	_ context.Context,
+	data *schema.ResourceData,
+	_ interface{},
+) ([]*schema.ResourceData, error) {
+	rawID := data.Id()
+	if rawID == "" {
+		return nil, fmt.Errorf("ID cannot be empty")
+	}
+
+	if !strings.Contains(rawID, "::") {
```

Comment:
In all the other resource pairs we use ":" as the separator, what made us use "::" over here? 🤔 Is it some kind of limitation when used with the user ID?

Reply:
Ouch 😢 , we have all the other resource pair getting imported just through ":", it would be great if we could make all of them consistent. Perhaps introduce another separator? And gate it under the v1 if we need a breaking change? Or make this one a bit smarter when it parses permission names?

Reply:
I'm ok to use `::` for all or only special cases but would like to avoid introducing this breaking change into v0.